src/tests/class_tests/openms/source/TheoreticalSpectrumGenerator_test.cpp

```diff
@@ -471,30 +471,34 @@ START_SECTION(static MSSpectrum generateSpectrum(const Precursor::ActivationMeth
 
   // Test ECD/ETD
   spec = TheoreticalSpectrumGenerator::generateSpectrum(Precursor::ActivationMethod::ECD, AASequence::fromString("HFYLWCP"), 1);
-  ABORT_IF(spec.size() != 11);
-  TEST_REAL_SIMILAR(spec[0].getPosition()[0], 99.044);
-  TEST_REAL_SIMILAR(spec[1].getPosition()[0], 202.0532);
-  TEST_REAL_SIMILAR(spec[2].getPosition()[0], 302.1611);
-  TEST_REAL_SIMILAR(spec[3].getPosition()[0], 388.1325);
-  TEST_REAL_SIMILAR(spec[4].getPosition()[0], 465.2244);
-  TEST_REAL_SIMILAR(spec[5].getPosition()[0], 501.2166);
-  TEST_REAL_SIMILAR(spec[6].getPosition()[0], 578.3085);
-  TEST_REAL_SIMILAR(spec[7].getPosition()[0], 664.2799);
-  TEST_REAL_SIMILAR(spec[8].getPosition()[0], 764.3878);
-  TEST_REAL_SIMILAR(spec[9].getPosition()[0], 811.3483);
-  TEST_REAL_SIMILAR(spec[10].getPosition()[0], 867.397);
+  TEST_EQUAL(spec.size(), 17);
+
+  TEST_REAL_SIMILAR(spec[0].getPosition()[0], 100.0518816);
+  TEST_REAL_SIMILAR(spec[1].getPosition()[0], 101.0597067);
+  TEST_REAL_SIMILAR(spec[2].getPosition()[0], 203.0610665);
+  TEST_REAL_SIMILAR(spec[3].getPosition()[0], 204.0688916);
+  TEST_REAL_SIMILAR(spec[4].getPosition()[0], 302.1611520);
+  TEST_REAL_SIMILAR(spec[5].getPosition()[0], 389.1403798);
+  TEST_REAL_SIMILAR(spec[6].getPosition()[0], 390.1482049);
+  TEST_REAL_SIMILAR(spec[7].getPosition()[0], 465.2244813);
+  TEST_REAL_SIMILAR(spec[8].getPosition()[0], 502.2244442);
+  TEST_REAL_SIMILAR(spec[9].getPosition()[0], 503.2322692);
+  TEST_REAL_SIMILAR(spec[10].getPosition()[0], 578.3085457);
+  // ...
 
   spec.clear(true);
 
   // Test precursor_charge > 2
   spec = TheoreticalSpectrumGenerator::generateSpectrum(Precursor::ActivationMethod::HCID, AASequence::fromString("PEP"), 3);
-  ABORT_IF(spec.size() != 6);
+  TEST_EQUAL(spec.size(), 8);
```

Comment:
what are the new ion types here?

Reply:
a-ions right?

Reply:
Yes should be the a-ions that are more likely in HCID, though still not as common as b/y.
This is debatable if you want them here or not. I thought rather too many than too few.

Reply:
They are kind of rare and I usually only observe partial coverage (e.g. just a2,a3). But this is also very much dependent on collision energy and data.
But I think your approach of creating more ions is sensible.
Someone who writes a e.g. search engine probably wants to specify ion types manually anyway and not via this enum. So probably not such an issue...